src/os/bluestore/BitAllocator.h

```diff
@@ -246,26 +246,20 @@ class BitMapArea {
 class BitMapAreaList {
 
 private:
-  BitMapArea **m_items;
-  int64_t m_num_items;
+  std::vector<BitMapArea*> m_items;
```

Comment:
Such an implementation (BitMapAreaList encapsulating a vector) seems excessive to me. IMO it brings nothing usefull. 
What's about making a typedef and cleanup corresponding access points, e.g. get_nth_item?